docker/docker_image_dest.go

```diff
@@ -291,25 +301,35 @@ func (d *dockerImageDestination) TryReusingBlob(ctx context.Context, info types.
 	if info.Digest == "" {
 		return false, types.BlobInfo{}, errors.Errorf(`"Can not check for a blob with unknown digest`)
 	}
+	bic := blobinfocache.FromBlobInfoCache(cache)
 
 	// First, check whether the blob happens to already exist at the destination.
 	exists, size, err := d.blobExists(ctx, d.ref.ref, info.Digest, nil)
 	if err != nil {
 		return false, types.BlobInfo{}, err
 	}
 	if exists {
-		cache.RecordKnownLocation(d.ref.Transport(), bicTransportScope(d.ref), info.Digest, newBICLocationReference(d.ref))
-		return true, types.BlobInfo{Digest: info.Digest, Size: size}, nil
+		bic.RecordKnownLocation2(d.ref.Transport(), bicTransportScope(d.ref), info.Digest, blobinfocache.UnknownCompression, newBICLocationReference(d.ref))
```

Comment:
Most of the time (but not for schema1), `info.MediaType` should be set and valid here. Recording `Unknown` and preventing the loop below from benefiting is not ideal.

So it would be useful to implement the MIME type → compression, or MIME type → [compression] → MIME type, lookup — and at that would in turn allow moving this out of the transports into the `TryReusingBlob` caller.

… except that we might not trust the contents of a manifest. Ugh.  But, see elsewhere, if we split recording the location from recording the compression, generic code can reliably detect compression and record it.